homeassistant/components/plugwise/climate.py

```diff
@@ -0,0 +1,310 @@
+    """Return active schema if present."""
+    domain_objects = self._api.get_domain_objects()
+    return self._api.get_active_schema_name(domain_objects)
+
+
+def _heating_action(self):
+    """Return current active heating state."""
+    domain_objects = self._api.get_domain_objects()
+    if self._api.get_heating_status(domain_objects) is True:
+        self._state = CURRENT_HVAC_HEAT
+    else:
+        self._state = CURRENT_HVAC_IDLE
+    return self._state
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Add the Plugwise (Anna) Thermostate."""
+    add_devices([
+        ThermostatDevice(
+            config.get(CONF_NAME),
+            config.get(CONF_USERNAME),
+            config.get(CONF_PASSWORD),
+            config.get(CONF_HOST),
+            config.get(CONF_PORT),
+            config.get(CONF_MIN_TEMP),
+            config.get(CONF_MAX_TEMP),
+        )
+    ])
+
+
+class ThermostatDevice(ClimateDevice):
+    """Representation of an Plugwise thermostat."""
+
+    def __init__(self, name, username, password, host, port,
+                 min_temp, max_temp):
+        """Set up the Plugwise API."""
+        _LOGGER.debug("Init called")
+        self._name = name
+        self._username = username
+        self._password = password
+        self._host = host
+        self._port = port
+        self._temperature = None
+        self._current_temperature = None
+        self._outdoor_temperature = None
+        self._state = None
+        self._active_schema = None
+        self._previous_schema = None
+        self._preset_mode = None
+        self._preset_modes = []
+        self._min_temp = min_temp
+        self._max_temp = max_temp
+        self._hvac_modes = ATTR_HVAC_MODES
+        self._hvac_mode = None
+
+        _LOGGER.debug("Initializing API")
+        import haanna
+        self._api = haanna.Haanna(self._username, self._password,
+                                  self._host, self._port)
+        try:
+            self._api.ping_anna_thermostat()
+        except Exception:
+            _LOGGER.error("Unable to ping, platform not ready")
+            raise PlatformNotReady
+        _LOGGER.debug("platform ready")
+        self.update()
+
+    @property
+    def should_poll(self):
+        """Return that we will require polling."""
+        return True
+
+    @property
+    def state(self):
+        """Return the current state."""
+        return self._state
+
+    @property
+    def name(self):
+        """Return the name of the thermostat, if any."""
+        return self._name
+
+    @property
+    def icon(self):
+        """Return the icon to use in the frontend."""
+        return DEFAULT_ICON
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_FLAGS
+
+    @property
+    def device_state_attributes(self):
+        """Return the device specific state attributes."""
+        _attributes = {}
```

Comment:
No need for _ prefix its a local scope variable.

Reply:
Adjusted accordingly